src/Faker/Provider/sv_SE/Person.php

```diff
@@ -132,17 +132,30 @@ public function personalIdentityNumber(\DateTime $birthdate = null, $gender = nu
             $birthdate = \Faker\Provider\DateTime::dateTimeThisCentury();
         }
         $datePart = $birthdate->format('ymd');
-
-        if ($gender && $gender == static::GENDER_MALE) {
-            $randomDigits = (string) static::numerify('##') . static::randomElement([1, 3, 5, 7, 9]);
-        } elseif ($gender && $gender == static::GENDER_FEMALE) {
-            $randomDigits = (string) static::numerify('##') . static::randomElement([0, 2, 4, 6, 8]);
-        } else {
-            $randomDigits = (string) static::numerify('###');
-        }
+        $randomDigits = $this->getBirthNumber($gender);
 
         $checksum = Luhn::computeCheckDigit($datePart . $randomDigits);
 
         return $datePart . '-' . $randomDigits . $checksum;
     }
+
+    /**
+     * @param string $gender Person::GENDER_MALE || Person::GENDER_FEMALE
+     *
+     * @return string of three digits
+     */
+    protected function getBirthNumber($gender = null)
+    {
+        do {
+            if ($gender && $gender == static::GENDER_MALE) {
+                $randomDigits = (string) static::numerify('##') . static::randomElement([1, 3, 5, 7, 9]);
+            } elseif ($gender && $gender == static::GENDER_FEMALE) {
+                $randomDigits = (string) static::numerify('##') . static::randomElement([0, 2, 4, 6, 8]);
+            } else {
+                $randomDigits = (string) static::numerify('###');
```

Comment:
Could tidy this up with an early return `if (!$gender` to make it a little more readable

Reply:
Correct, unwanted and unneeded else statement.